stringtie/1.2.4/prepDE.py

```diff
@@ -61,9 +67,14 @@ def t_overlap(t1, t2): #from badGenes: chromosome, strand, cluster, start, end,
 
 for s in samples:
     badGenes=[] #list of bad genes (just ones that aren't MSTRG)
-    
-    try:
-        with open(glob.iglob(os.path.join(opts.input,s,"*.gtf")).next()) as f:
+
+    # Find a GTF file in the sample directory.
+    sample_path = os.path.join(opts.input, s)
+    gtf_file = next(iter(list_gtf_files(sample_path)), None)
+
+    # Build the list of bad genes if the GTF file is present.
+    if gtf_file is not None:
+        with open(gtf_file) as f:
```

Comment:
The `try` block was there only to catch `StopIteration` exceptions so that the script could go to the next sample directory if such an exception was caught. The new function doesn't rely on the Exception, so the try block is no longer necessary.